packages/documentation/public/assets/scripts/storybook-preview-events.js

```diff
@@ -0,0 +1,30 @@
+const previewIframe = document.body;
+let footerEl = document.querySelector('.docs-footer');
+let currentPage = null;
+
+if (footerEl) {
+  // if footer already exists, just emit ready event and listen for route-changes
+  contentReady();
+} else {
+  // if footer does not exist yet, wait until its rendered, then emit contentReady event and listen for route-changes
+  new MutationObserver(function () {
```

Comment:
Unfortunately, we have to make all this code to wait until the footer is available. Maybe after https://github.com/storybookjs/storybook/issues/24429, we could simply put the script at the end of the body and simplify it?